### ISSUE TITLE
Feature request: custom path to TF graph file in NerDLApproach

### ISSUE THREAD

**Issue by maziyarpanahi** [JohnSnowLabs/spark-nlp#507]
Currently, the only way to use custom generated TensorFlow graph is to add it to the default path `spark-nlp\src\main\resources\ner-dl` and re-build the JAR. This also blocks using a custom TF graph with Spark Packages (`--packages`) since the JAR is coming with the default TF graph files.

I suggest adding a parameter that accepts a path to a TF graph file generated by the user. If this param is not set it can look into the default path.
Something like: `setGraphFilePath`

**Comment by Tshimanga** [comment]
Beyond loading custom graph for training via `NerDLApproach`, it would be great to be able to load a pre-trained TF model directly into the `NerDLModel`. That way, I could train TF models in the comfort of the existing and mature TF ecosystem and then still deploy to a spark pipeline when I'm done.

**Comment by saifjsl** [comment]
> 
> 
> Beyond loading custom graph for training via `NerDLApproach`, it would be great to be able to load a pre-trained TF model directly into the `NerDLModel`. That way, I could train TF models in the comfort of the existing and mature TF ecosystem and then still deploy to a spark pipeline when I'm done.

Wouldn't that be a TF Wrapper annotator? in case of NER, your TF model would need to have specific set of layers and nodes to be a compatible one. And a contract in the annotator wrapper that reads generic information that could be applied from any TF model, right? 

If I read you correctly, there's no need to make it a `NerDLModel` at all

**Comment by Tshimanga** [comment]
Since the `NerDLModel` doesn't make any assertions about what and how many labels you are predicting with, it supports a wider variety of tasks that fall under the more general classification "sequence tagging", such as the deep sentence splitting, negation detection, etc. since the class has been demonstrated as general to any "sequence of tokens" -> "sequence of annotated tokens" task, it'd be nice to accommodate any suitable model.

For instance, if I train a tf model that performs sentence splitting with a vanilla RNN with 1 layer and 300 nodes or a full hierarchical model like the charLSTM-biLSTM-CRF I'd like be able to point to either so long as they both satisfy the interface of "give me embeddings and tokens, and ill give you tagged tokens". It would greatly reduce the barrier to using SparkNLP as a deployment framework.
